### PR TITLE
clang-tidy: apply modernize-loop-convert over a set of files 

### PR DESCRIPTION
**Pull request by andy31415**
Applied a `modernize-loop-convert` using chip-tool, all-clusters and tests as the sources of files to compile.

### CODE REVIEW

File: examples/chip-tool/commands/common/Command.cpp [hunk 1]
```diff
@@ -877,9 +877,8 @@ void ResetOptionalArg(const Argument & arg)
 
 void Command::ResetArguments()
 {
-    for (size_t i = 0; i < mArgs.size(); i++)
+    for (auto arg : mArgs)
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/common/Command.cpp:880]: Can this be `const auto & arg`?

File: examples/chip-tool/commands/common/Command.cpp [hunk 2]
```diff
@@ -42,9 +42,9 @@ bool Command::InitArguments(int argc, char ** argv)
     size_t argvExtraArgsCount = (size_t) argc;
     size_t mandatoryArgsCount = 0;
     size_t optionalArgsCount  = 0;
-    for (size_t i = 0; i < mArgs.size(); i++)
+    for (auto & mArg : mArgs)
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/common/Command.cpp:45]:
```suggestion
    for (auto & arg : mArgs)
```

File: src/app/BufferedReadCallback.cpp [hunk 3]
```diff
@@ -65,9 +65,9 @@ CHIP_ERROR BufferedReadCallback::GenerateListTLV(TLV::ScopedBufferTLVReader & aR
     // To avoid that, a single contiguous buffer is the best likely approach for now.
     //
     uint32_t totalBufSize = 0;
-    for (size_t i = 0; i < mBufferedList.size(); i++)
+    for (auto & i : mBufferedList)
```
Comment (bzbarsky-apple) [inline src/app/BufferedReadCallback.cpp:68]: That's really odd variable naming...  And does this need to be a non-const ref?

File: src/app/tests/TestPendingNotificationMap.cpp [hunk 4]
```diff
@@ -80,10 +80,10 @@ void TestAddRemove(nlTestSuite * aSuite, void * aContext)
     pendingMap.RemoveAllEntriesForNode(chip::ScopedNodeId());
     uint8_t expectedEntryIndecies[] = { 1, 2, 3, 4, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19 };
     iter                            = pendingMap.begin();
-    for (size_t i = 0; i < sizeof(expectedEntryIndecies); i++)
+    for (unsigned char expectedEntryIndecie : expectedEntryIndecies)
```
Comment (bzbarsky-apple) [inline src/app/tests/TestPendingNotificationMap.cpp:83]:
```suggestion
    for (unsigned char expectedEntryIndex : expectedEntryIndices)
```
Part of the miss-spelling is pre-existing, but the automated conversion just made it even worse.

File: src/ble/BleLayer.cpp [hunk 5]
```diff
@@ -203,9 +203,9 @@ CHIP_ERROR BleTransportCapabilitiesRequestMessage::Decode(const PacketBufferHand
     VerifyOrReturnError(CAPABILITIES_MSG_CHECK_BYTE_1 == chip::Encoding::Read8(p), BLE_ERROR_INVALID_MESSAGE);
     VerifyOrReturnError(CAPABILITIES_MSG_CHECK_BYTE_2 == chip::Encoding::Read8(p), BLE_ERROR_INVALID_MESSAGE);
 
-    for (size_t i = 0; i < kCapabilitiesRequestSupportedVersionsLength; i++)
+    for (unsigned char & mSupportedProtocolVersion : msg.mSupportedProtocolVersions)
```
Comment (bzbarsky-apple) [inline src/ble/BleLayer.cpp:206]:
```suggestion
    for (unsigned char & supportedProtocolVersion : msg.mSupportedProtocolVersions)
```
But also, what guarantees that `sizeof(msg.mSupportedProtocolVersions) == kCapabilitiesRequestSupportedVersionsLength`?  They are not obviously the same size.  We need a static_assert at least.

File: src/controller/tests/TestWriteChunking.cpp [hunk 6]
```diff
@@ -289,9 +289,9 @@ void TestWriteChunking::TestBadChunking(nlTestSuite * apSuite, void * apContext)
         app::WriteClient writeClient(&ctx.GetExchangeManager(), &writeCallback, Optional<uint16_t>::Missing());
 
         ByteSpan list[kTestListLength];
-        for (uint8_t j = 0; j < kTestListLength; j++)
+        for (auto & j : list)
```
Comment (bzbarsky-apple) [inline src/controller/tests/TestWriteChunking.cpp:292]:
```suggestion
        for (auto & item : list)
```

File: src/credentials/CHIPCert.cpp [hunk 7]
```diff
@@ -616,9 +616,9 @@ ChipDN::~ChipDN() {}
 
 void ChipDN::Clear()
 {
-    for (uint8_t i = 0; i < CHIP_CONFIG_CERT_MAX_RDN_ATTRIBUTES; i++)
+    for (auto & i : rdn)
```
Comment (bzbarsky-apple) [inline src/credentials/CHIPCert.cpp:619]:
```suggestion
    for (auto & dn : rdn)
```

File: src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp [hunk 8]
```diff
@@ -645,11 +645,11 @@ CHIP_ERROR CsaCdKeysTrustStore::AddTrustedKey(const ByteSpan & derCertBytes)
 CHIP_ERROR CsaCdKeysTrustStore::LookupVerifyingKey(const ByteSpan & kid, Crypto::P256PublicKey & outPubKey) const
 {
     // First, search for the well known keys
-    for (size_t keyIdx = 0; keyIdx < gCdSigningKeys.size(); keyIdx++)
+    for (auto & gCdSigningKey : gCdSigningKeys)
```
Comment (bzbarsky-apple) [inline src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp:648]:
```suggestion
    for (auto & cdSigningKey : gCdSigningKeys)
```

File: src/credentials/tests/TestChipCert.cpp [hunk 9]
```diff
@@ -1137,11 +1133,8 @@ static void TestChipCert_CertType(nlTestSuite * inSuite, void * inContext)
         {  TestCert::kNode02_02,       kCertType_Node            },
     };
     // clang-format on
-    static const size_t sNumTestCases = ArraySize(sTestCases);
-
-    for (unsigned i = 0; i < sNumTestCases; i++)
+    for (auto testCase : sTestCases)
```
Comment (bzbarsky-apple) [inline src/credentials/tests/TestChipCert.cpp:1136]: Why not a const ref?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
I only got partway through, but there are a lot of issues here in terms of the naming and type selection for the loops....

File: examples/chip-tool/commands/common/Command.cpp (hunk 1 above)
Comment (andy31415) [reply examples/chip-tool/commands/common/Command.cpp:880]: Will go through renames manually and update the names. Will update this to const as well.

File: src/app/BufferedReadCallback.cpp (hunk 3 above)
Comment (andy31415) [reply src/app/BufferedReadCallback.cpp:68]: Automated tool seems limited. Will  manually go through things.

File: src/app/tests/TestPendingNotificationMap.cpp (hunk 4 above)
Comment (andy31415) [reply src/app/tests/TestPendingNotificationMap.cpp:83]: renamed to `ch`

File: src/ble/BleLayer.cpp (hunk 5 above)
Comment (andy31415) [reply src/ble/BleLayer.cpp:206]: I think this for loop conversion was wrong. Will revert it back.

Comment (andy31415) [reply src/ble/BleLayer.cpp:206]: Actually added the assert - I was looking at a different constant when comparing the logic. The automated conversion seems accurate.

**Comment by andy31415** [comment]
> I only got partway through, but there are a lot of issues here in terms of the naming and type selection for the loops....

@bzbarsky-apple went manually through the changes and used more sane names. Also converted a setter to take a const array to make a const-correctness test better. Overall same change, but with saner names now I hope. Did not try hard for converting auto to const auto &, however some where converted. I am going for "this is slightly better than before".

**Review by tehampson — APPROVED** [review]

**Comment by tehampson** [comment]
Do we want to update `.clang-tidy`? Or was this just a partial update?

**Comment by andy31415** [comment]
> Do we want to update `.clang-tidy`? Or was this just a partial update?

I was unsure if forcing to use loop is something that we want everywhere. My instinct was that this does not always feel natural, but at the same time I 100% accepted all automated changes. Would welcome opinions from reviewers here if we should change clang-tidy to enforce or not.

File: src/app/BufferedReadCallback.cpp [hunk 10]
```diff
@@ -65,9 +65,9 @@ CHIP_ERROR BufferedReadCallback::GenerateListTLV(TLV::ScopedBufferTLVReader & aR
     // To avoid that, a single contiguous buffer is the best likely approach for now.
     //
     uint32_t totalBufSize = 0;
-    for (size_t i = 0; i < mBufferedList.size(); i++)
+    for (const auto & list : mBufferedList)
```
Comment (bzbarsky-apple) [inline src/app/BufferedReadCallback.cpp:68]:
```suggestion
    for (const auto & packetBuffer : mBufferedList)
```

File: src/app/tests/TestPendingNotificationMap.cpp [hunk 11]
```diff
@@ -80,10 +80,10 @@ void TestAddRemove(nlTestSuite * aSuite, void * aContext)
     pendingMap.RemoveAllEntriesForNode(chip::ScopedNodeId());
     uint8_t expectedEntryIndecies[] = { 1, 2, 3, 4, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19 };
     iter                            = pendingMap.begin();
-    for (size_t i = 0; i < sizeof(expectedEntryIndecies); i++)
+    for (unsigned char ch : expectedEntryIndecies)
```
Comment (bzbarsky-apple) [inline src/app/tests/TestPendingNotificationMap.cpp:83]:
```suggestion
    for (uint8_t ch : expectedEntryIndecies)
```

File: src/lib/asn1/ASN1OID.cpp [hunk 12]
```diff
@@ -46,11 +46,11 @@ DLL_EXPORT OID ParseObjectID(const uint8_t * encodedOID, uint16_t encodedOIDLen)
         return kOID_NotSpecified;
     }
 
-    for (uint32_t i = 0; i < sOIDTableSize; i++)
+    for (const auto & oid : sOIDTable)
```
Comment (bzbarsky-apple) [inline src/lib/asn1/ASN1OID.cpp:49]: The use of different names in the different loops in this file is a bit odd, but ok.

File: src/lib/dnssd/tests/TestTxtFields.cpp [hunk 13]
```diff
@@ -293,9 +293,9 @@ bool NodeDataIsEmpty(const DiscoveredNodeData & node)
     {
         return false;
     }
-    for (size_t i = 0; i < sizeof(CommissionNodeData::rotatingId); ++i)
+    for (unsigned char id : node.commissionData.rotatingId)
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/tests/TestTxtFields.cpp:296]:
```suggestion
    for (uint8_t id : node.commissionData.rotatingId)
```

File: src/system/tests/TestSystemPacketBuffer.cpp [hunk 14]
```diff
@@ -1817,9 +1817,9 @@ void PacketBufferTest::CheckHandleCloneData(nlTestSuite * inSuite, void * inCont
     NL_TEST_ASSERT(inSuite, test->mContext == theContext);
 
     uint8_t lPayload[2 * PacketBuffer::kMaxSizeWithoutReserve];
-    for (size_t i = 0; i < sizeof(lPayload); ++i)
+    for (unsigned char & payload : lPayload)
```
Comment (bzbarsky-apple) [inline src/system/tests/TestSystemPacketBuffer.cpp:1820]:
```suggestion
    for (uint8_t & byte : lPayload)
```

File: src/transport/tests/TestSessionManager.cpp [hunk 15]
```diff
@@ -816,17 +816,17 @@ void SessionAllocationTest(nlTestSuite * inSuite, void * inContext)
                 ScopedNodeId(NodeIdFromPAKEKeyId(kDefaultCommissioningPasscodeId), kUndefinedFabricIndex));
             NL_TEST_ASSERT(inSuite, handle.HasValue());
             auto potentialCollision = handle.Value()->AsSecureSession()->GetLocalSessionId();
-            for (size_t h = 0; h < numHandles; ++h)
+            for (unsigned short sessionId : sessionIds)
```
Comment (bzbarsky-apple) [inline src/transport/tests/TestSessionManager.cpp:819]:
```suggestion
            for (uint16_t sessionId : sessionIds)
```

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**